bundles/mapping/mapwfs2/domain/WfsLayerModelBuilder.js

```diff
@@ -5,32 +5,85 @@
 Oskari.clazz.define(
     'Oskari.mapframework.bundle.mapwfs2.domain.WfsLayerModelBuilder',
 
-    function (sandbox) {
+    function(sandbox) {
         this.localization = Oskari.getLocalization('MapWfs2');
         this.sandbox = sandbox;
+        this.service = null;
     }, {
+
+        /**
+         * @private @method _checkIfAggregateValuesAreAvailable
+         * function gives value to addLinkToAggregateValues (true/false)
+         *
+         * @return {Boolean}
+         */
+        _checkIfAggregateValuesAreAvailable: function() {
+            this.service = this.sandbox.getService(
+                'Oskari.analysis.bundle.analyse.service.AnalyseService'
+            );
+            if (!this.service) {
+                return false;
+            }
+            return true;
+        },
         /**
          * parses any additional fields to model
          * @param {Oskari.mapframework.domain.WfsLayer} layer partially populated layer
          * @param {Object} mapLayerJson JSON presentation of the layer
          * @param {Oskari.mapframework.service.MapLayerService} maplayerService not really needed here
          */
-        parseLayerData: function (layer, mapLayerJson, maplayerService) {
-            var me = this,
-                toolBuilder = Oskari.clazz.builder('Oskari.mapframework.domain.Tool');
+        parseLayerData: function(layer, mapLayerJson, maplayerService) {
+            var me = this;
 
             if (layer.isLayerOfType("WFS")) {
-                var locOwnStyle = this.localization['own-style'],
-                    toolOwnStyle = toolBuilder();
+                var locOwnStyle = me.localization['own-style'];
+                var toolOwnStyle = Oskari.clazz.create('Oskari.mapframework.domain.Tool');
                 toolOwnStyle.setName("ownStyle");
                 toolOwnStyle.setTitle(locOwnStyle);
+                toolOwnStyle.setIconCls('show-own-style-tool');
                 toolOwnStyle.setTooltip(locOwnStyle);
-                toolOwnStyle.setCallback(function () {
+                toolOwnStyle.setCallback(function() {
                     me.sandbox.postRequestByName('ShowOwnStyleRequest', [layer.getId()]);
                 });
                 layer.addTool(toolOwnStyle);
             }
 
+            if (layer.isLayerOfType("WFS") || layer.isLayerOfType("ANALYSIS")) {
```

Comment:
Analysislayers has it's own modelbuilder: https://github.com/oskariorg/oskari-frontend/blob/master/bundles/mapping/mapanalysis/domain/AnalysisLayerModelBuilder.js move the analysis stuff there.

Reply:
Or maybe do something like layer.isFilterSupported() that returns true for WFS and ANALYSIS layers, but not for myplaces or userlayers.